shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -446,41 +446,76 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
+                    // Execute the prelude code.
                     prelude match {
-                      case Nil => ()
-                      case lines => host.execute(lines mkString " ")
+                      case Nil => {
+                        if (mode.showRepl) {
+                          output(s"├── No prelude")
+                          if (queries.isEmpty)
+                            output(s"└── No queries")
+                        }
+                      }
+                      case lines => {
+                        val preludeReply = host.execute(lines mkString " ")
+                        if (mode.showRepl) {
+                          output(s"├─┬ Prelude")
+                          output(s"│ ├── Code")
+                          lines.iterator.foreach { line => output(s"│ │   $line") }
+                          // Display successful results in multiple lines.
+                          // Display other results in single line.
+                          preludeReply match {
+                            case ReplHost.Result(content, intermediate) =>
+                              intermediate.foreach { value =>
+                                output(s"│ ├── Intermediate")
+                                content.linesIterator.foreach { line => output(s"│ │   $line") }  
+                              }
+                              output(s"│ └── Reply")
+                              content.linesIterator.foreach { line => output(s"│     $line") }  
+                            case other => output(s"│ └── Reply $other")
+                          }
+                        }
+                      }
                     }
-                    if (mode.showRepl) {
-                      println(s"Block [line: ${blockLineNum}] [file: ${file.baseName}]")
-                      if (queries.isEmpty)
-                        println(s"The block is empty")
+                    if (mode.showRepl && queries.isEmpty) {
+                      output(s"└── No queries")
                     }
                     // Send queries to the host.
                     ExecutedResult(queries.zipWithIndex.map {
                       case (JSTestBackend.CodeQuery(preludeLines, codeLines, res), i) =>
                         val prelude = preludeLines.mkString
                         val code = codeLines.mkString
+                        val p0 = if (i + 1 == queries.length) "  " else "│ "
                         if (mode.showRepl) {
-                          println(s"├── Query ${i + 1}/${queries.length}")
-                          println(s"│ ├── Prelude: ${if (preludeLines.isEmpty) "<empty>" else prelude}")
-                          println(s"│ └── Code: ${code}")
+                          val p1 = if (i + 1 == queries.length) "└─" else "├─"
+                          output(s"$p1┬ Query ${i + 1}/${queries.length}")
+                          output(s"$p0├── Prelude: ${if (preludeLines.isEmpty) "<empty>" else prelude}")
```

Comment:
It seems that this is always empty in practice (see tests). If it is, better remove this stuff. Otherwise, when is it not empty? Can we have a test for it?

Reply:
I added a test case indicating its usage in 5bf8d35.